nixos/modules/misc/version.nix

```diff
@@ -102,6 +103,8 @@ in
       # changing them would not rebuild the manual
       nixosLabel   = mkDefault (maybeEnv "NIXOS_LABEL" cfg.nixosVersion);
       nixosVersion = mkDefault (maybeEnv "NIXOS_VERSION" (cfg.nixosRelease + cfg.nixosVersionSuffix));
+      nixosRevision      = mkIf (pathExists gitRepo) (mkDefault            (lib.substring 0 7 (commitIdFromGitRepo gitRepo)));
+      nixosVersionSuffix = mkIf (pathExists gitRepo) (mkDefault (".git." + (lib.substring 0 7 (commitIdFromGitRepo gitRepo))));
```

Comment:
I would factor out `lib.substring 0 7 (commitIdFromGitRepo gitRepo)` into the let above, as calling this new function can be relatively expensive so we would better ensure it's not done twice.